test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceMappingConfigurationTests.cs

```diff
@@ -85,5 +85,91 @@ public void GetPackageSourceMappingConfiguration_WithMultipleSources()
             IReadOnlyList<string> contosoPattern = packageSourceMapping.Patterns["contoso"];
             contosoPattern.Should().BeEquivalentTo(new string[] { "moreStuff" });
         }
+
+        [Fact]
+        public void GetPackageSourceMappingConfiguration_WithMultipleConfigs_WhenMatchingKeys_PicksClosestOne()
```

Comment:
This is great behavior to test. Is it related to the case sensitivity or is it a bigger behavior change in NuGet.Configuration?

Reply:
I'm just adding a test confirming the expectation that the merging is done based on an ignore case key. 
There's no changes in behavior, just something that wasnt't covered.